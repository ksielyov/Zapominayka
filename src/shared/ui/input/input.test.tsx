import React from 'react';
import renderer from 'react-test-renderer';
import {Input} from '@ui';
import {fireEvent, render} from '@testing-library/react-native';

test('Input should render correctly', () => {
  const tree = renderer
    .create(<Input placeholder={'Input test'} onChangeText={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input should change value to `Input test`', () => {
  const onChangeTextMock = jest.fn();
  const testValue = 'test value';
  const placeholderText = 'Input test';

  const {getByPlaceholderText} = render(
    <Input onChangeText={onChangeTextMock} placeholder={placeholderText} />,
  );

  fireEvent.changeText(getByPlaceholderText(placeholderText), testValue);
  expect(onChangeTextMock).toHaveBeenCalledWith(testValue);
});
