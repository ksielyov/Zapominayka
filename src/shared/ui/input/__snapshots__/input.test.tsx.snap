// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 11.770588235294117,
      "height": 86.31764705882352,
      "width": "100%",
    }
  }
>
  <TextInput
    placeholder="Input test"
    placeholderTextColor="rgba(21,22,37,0.6)"
    style={
      Object {
        "color": "#151625",
        "fontSize": 26.68,
        "height": "100%",
        "paddingHorizontal": 30,
        "width": "100%",
      }
    }
    textAlignVertical="center"
  />
</View>
`;
